cluster-autoscaler/processors/podinjection/pod_injection_processor.go

```diff
@@ -61,9 +62,16 @@ func (p *PodInjectionPodListProcessor) Process(ctx *context.AutoscalingContext,
 	}
 	scheduledPods := podsFromNodeInfos(nodeInfos)
 
-	groupedPods := groupPods(append(scheduledPods, unschedulablePods...), controllers)
-	var podsToInject []*apiv1.Pod
+	allPods, err := ctx.AllPodLister().List()
+	if err != nil {
+		klog.Errorf("Failed to list all pods from all pod lister: %v", err)
```

Comment:
This is redundant with the error returned - we'll end up logging the error twice. I'd just return the error and let the caller do the logging.